### ISSUE TITLE
core.ask in double nested mistral workflow sometimes fails to set the status of root workflow to paused

### ISSUE THREAD

**Issue by shusugmt** [StackStorm/st2#4138]
##### ISSUE TYPE
 - Bug Report

##### STACKSTORM VERSION
```
st2 2.7.2, on Python 2.7
```

##### OS / ENVIRONMENT / INSTALL METHOD
- st2-docker

##### SUMMARY
This is related to #4135 and #4137 
Assume we have following mistral workflow:

```
root_workflow
  - nested_workflow
    - action1
    - action2
    - core.ask
    - action3
```

In `core.ask`, it will set the status of *parent* workflow = `nested_workflow` in this case, to `paused`. As a result, the root_workflow also changes to `paused` state.

But sometimes the root_workflow remains in `pausing` state. As far as I can see, it is just random. Some executions gets into paused, some won't and stuck in `pausing` state.  And it turns out that this seems to be the root cause of #4135

**Comment by dbillor** [comment]
Experiencing same issue.  Have to manually resume and track workflows down.  Hard to reproduce as it is transient.

**Comment by shusugmt** [comment]
This is solved by #4309
